machine_learning_hep/globalfitter.py

```diff
@@ -58,255 +28,280 @@ def fixpar(massmin, massmax, masspeak, range_signal):
-        print("begin fit")
-        if dolikelihood is True:
-            fitOption = "L,E"
-        if setinitialgaussianmean is True:
-            mean = masspeak
-        if setfixgaussiansigma is True:
-            fixedsigma == 1
-        sigmaSgn = sigma
-
-        print("fit background (just side bands)")
-        fitsidebands = True
-        nSigma4SideBands = 4.
-        range_signal = nSigma4SideBands*sigmaSgn
-        integralhisto = Double(histo.Integral(histo.FindBin(massmin), \
-                                              histo.FindBin(massmax), "width"))
-        back_fit = func_fit(fitsidebands, bkgfunc, massmin, massmax,\
-            integralhisto, masspeak, range_signal)
-        back_fit.SetLineColor(kBlue+3)
-        histo.Fit("back_fit", ("R,%s,+,0" % (fitOption)))
-        back_fit.SetLineColor(kGray+1)
-        back_fit.SetLineStyle(2)
+# pylint: disable=too-many-statements
+def fitter(histo, case, sgnfunc, bkgfunc, masspeak, rebin, dolikelihood, setinitialgaussianmean,
+           setfixgaussiansigma, sigma_sig, massmin, massmax, fixedmean, outputfolder, suffix,
+           draw_side_band_fit=False):
+    """
+    Some comments:
+        -> The object 'back_fit' is fitted using only the side-bands
+        -> The object 'tot_fit' is fitted to the entire extracted inv-mass distribution
+        -> The object 'back_refit' is not used for fitting but only stores the background fit
+           parameters from fitted 'tot_fit'
+        -> The object 'sig_fit' is not used for fitting but only stores the signal fit
+           parameters from fitted 'tot_fit'
+    """
+    logger = get_logger()
+
+    logger.info("Do fitting")
+    logger.info("Argument \"setinitialgaussianmean\" set to %u, however not used at the moment",
+                setinitialgaussianmean)
+
+    if "Lc" not in case and "D0" not in case and "Ds" not in case:
+        logger.warning("Can only do the fit for Lc, D0 or Ds, however found case %s", case)
+        return -1, -1
+
+    histo.GetXaxis().SetTitle("Invariant Mass L_{c}^{+}(GeV/c^{2})")
+    histo.Rebin(rebin)
+    histo.SetStats(0)
+    TVirtualFitter.SetDefaultFitter("Minuit")
+
+    # Prepare what can be prepared
+    npar_sig = 3
+    npar_bkg = 3 if bkgfunc == "Pol2" else 2
+    par_names_sig = ["IntSig", "Mean", "Sigma"]
+    par_names_bkg = ["IntBkg", "BkgCoeff1", "BkgCoeff2"] if bkgfunc == "Pol2" \
+                                                         else ["IntBkg", "BkgCoeff1"]
+
+    fitOption = "L,E" if dolikelihood else ""
+
+    logger.info("Initial parameters for signal fit are")
+    print(f"mean = {masspeak}\nsigma = {sigma_sig}")
+
+    logger.debug("fit background (just side bands)")
+    nSigma4SideBands = 4.
+    range_signal = nSigma4SideBands * sigma_sig
+    integralhisto = Double(histo.Integral(histo.FindBin(massmin), histo.FindBin(massmax), "width"))
+    back_fit = bkg_fit_func("bkg_fit_sidebands", bkgfunc, massmin, massmax, integralhisto,
```

Comment:
One of the most important corrections as there was no side-band fit in the old version (wrong function names "back_fit" vs. "back_fit_new")